### ISSUE TITLE
Push socket doesn't block when there are no Pull sockets listening

### ISSUE THREAD

**Issue by dirkcuys** [zeromq/zeromq.js#402]
**Describe the bug**
From the [documentation](https://zeromq.org/socket-api/) under the "Pipeline pattern" header:
> if there are no downstream nodes at all, then any send operations on the socket will block until the mute state ends or at least one downstream node becomes available for sending;

The socket doesn't block, but the node script then doesn't quit because there is still a listener for the send. 

**Reproducing**
```javascript
const zmq = require("zeromq");

(async () => {
  try {
    const reSock = new zmq.Push;
    reSock.connect('tcp://127.0.0.1:3002');
    await reSock.send(JSON.stringify({
      payload: 'this message goes nowhere',
    }))
    await reSock.close();
  } catch (e) {
    console.log(e);
  }
  console.log('done');
  // process.exit(1);
})();
```
My script prints "done" and then waits for a Pull socket to connect to.
> done
> (node:26664) WARNING: Waiting for queued ZeroMQ messages to be delivered. Set 'context.blocky = false' to change this behaviour.

The Warning only showed up after I upgraded from beta.5 t beta.6.

**Expected behavior**
From the documentation I understand it should block at `await reSock.send`

**Tested on**
 - OS: Ubuntu 18.04
 - ZeroMQ.js version: 6.0.0-beta.6
 - Node: 12.18.0

**Comment by rolftimmermans** [comment]
As soon as you `connect()` there will be a queue for that peer, so a send call will succeed if the high water mark is not reached. As far as I can see your example demonstrates the intended behaviour.

When the connection succeeds the messages will be queued.

The warning has been added because it can be very counter-intuitive in Node.js that ZMQ can prevent an otherwise graceful exit. During the period when queued messages are delivered in the background thread, the Node.js runtime has already shut down so no non-ZMQ work will be done by the process. This is not how most Node.js programs work, but it follows libzmq semantics; hence the warning.

**Comment by dirkcuys** [comment]
Thanks @rolftimmermans, I understood the documentation to mean that either the high water mark, or no connected PULL clients would cause send to block

> When a PUSH socket enters the mute state due to having reached the high water mark for all downstream nodes, or if there are no downstream nodes at all, then any send operations on the socket will block until the mute state ends or at least one downstream node becomes available for sending; messages are not discarded.

**Comment by rolftimmermans** [comment]
It's in the documentation for http://api.zeromq.org/4-3:zmq-connect:

>>> Following a zmq_connect(), for socket types except for ZMQ_ROUTER, the socket enters its normal ready state. By contrast, following a zmq_bind() alone, the socket enters a mute state in which the socket blocks or drops messages according to the socket type, as defined in zmq_socket(3).

**Comment by trusktr** [comment]
Hello! 

I was using zeromq 5, and I upgraded to 6-beta.6 yesterday. Now when I hit `ctrl-r` to refresh my `electron` application, it hangs indefinitely (have to exit with `ctrl-c`).

There is no output in devtools. But eventually I found that running `electron` with `ELECTRON_ENABLE_LOGGING=true` does show on last message in the terminal, which happens to be

```
(node:73342) WARNING: Waiting for queued ZeroMQ messages to be delivered. Set 'context.blocky = false' to change this behaviour.
```

I'm using `v5-compat`, so I think it should behave the same. I believe closing a process or refreshing an electron app which closes the node context should not block indefinitely.
